torch/utils/data/datapipes/utils/snapshot.py

```diff
@@ -0,0 +1,47 @@
+from torch.utils.data.datapipes.datapipe import IterDataPipe
+from torch.utils.data.graph_settings import apply_shuffle_seed
+
+__all__ = [
+    "simple_fast_forward_graph",
+]
+
+
+# TODO: Caveats
+#   1. Caller (either the ReadingService or DataLoader) must pass in the initial RNG
+#   2. `in_batch_shuffle` and `bucketbatch` are not compatible with this because they currently
+#      lack the option to `set_seed`.
+
+def simple_fast_forward_graph(datapipe: IterDataPipe, n_iterations: int, rng=None) -> None:
+    r"""
+    This function will fast-forward the given DataPipe by ``n_iterations``, and in the process,
+    fast-forward its parent DataPipes as well at the cost of re-doing every computation.
+    For instance, applying this function to the final DataPipe of a graph will fast-forward
+    every DataPipe within the graph.
+
+    Note:
+        This is the simplest but least efficient way to fast-forward a DataPipe. Usage of other fast-forwarding
+        methods (custom ones if necessary) are recommended.
+
+    Args:
+        datapipe: IterDataPipe to be fast-forwarded
+        n_iterations: number of iterations to fast-forward
+        rng: ``Optional[torch.Generator]``. If not ``None``, this RNG will be used for shuffling. The generator
+            should be in its `initial` state as it was first passed into ``DataLoader`` or ``ReadingService``.
+    """
+    apply_shuffle_seed(datapipe, rng)
+
```

Comment:
You should probably check if DP is already in iterator stage and prevent fast forwarding it. This will make sure that snapshot can be only restored to the 'new' graph.